File: shar/ws7_ls_wg/www/embed.ts
import * as fs from 'fs'
import * as ts from 'typescript'

const src = fs.readFileSync('index.ts', 'utf8')
const res = ts.transpileModule(src, {
    compilerOptions: {
        target: ts.ScriptTarget.ES2020,
    }
})

const js = Buffer.from(res.outputText)
const html = fs.readFileSync('index.html')

function toHex(buf: Buffer): string {
    return Array.from(buf).map(b => `0x${b.toString(16).padStart(2, '0')}`).join(', ')
}

const hpp = 
`#pragma once

const int index_html_size = ${html.length};
const unsigned char index_html[] = { ${toHex(html)} };

const int index_js_size = ${js.length};
const unsigned char index_js[] = { ${toHex(js)} };
`

fs.writeFileSync('../www.hpp', hpp)

File: shar/ws7_ls_wg/www/embed.py
with open('index.html', 'rb') as html:
    html = html.read()
with open('index.js', 'rb') as js:
    js = js.read()
with open('www.hpp', 'wt') as w:
    w.wreite('#pragma once\n')
    w.write(f'const int index_html_size = 0x{len(html):x};\n')
    w.write('const unsigned char index_html[] = { ')
    w.write(', '.join([f'0x{x:02x}' for x in html]))
    w.write(' };\n')
    
    w.write(f'const int index_js_size = 0x{len(js):x};\n')
    w.write('const unsigned char index_js[] = { ')
    w.write(', '.join([f'0x{x:02x}' for x in js]))
    w.write(' };\n')
